apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/ShardDownloader.java

```diff
@@ -174,8 +174,17 @@ public FileDownloadDecision prepareAndStartDownload() {
                 return result;
             }
             // prepare downloading
+            Set<Peer> lastShardPeers = shardsPeers.get(lastShard);
```

Comment:
The same issue is here. What is 'lastShard' is NULL ?

Reply:
It can not be beasuse it is not empty, see above